src/udev/net/ethtool-util.c

```diff
@@ -665,3 +693,48 @@ int ethtool_set_channels(int *fd, const char *ifname, netdev_channels *channels)
 
         return 0;
 }
+
+int config_parse_advertise(const char *unit,
+                           const char *filename,
+                           unsigned line,
+                           const char *section,
+                           unsigned section_line,
+                           const char *lvalue,
+                           int ltype,
+                           const char *rvalue,
+                           void *data,
+                           void *userdata) {
+        link_config *config = data;
+        NetDevAdvertise mode;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        for (;;) {
+                _cleanup_free_ char *w = NULL;
+
+                r = extract_first_word(&rvalue, &w, NULL, 0);
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to split advertise modes, ignoring: %s", rvalue);
+                        break;
+                }
+                if (r == 0)
+                        break;
+
+                mode = advertise_from_string(w);
+                if (mode == _NET_DEV_ADVERTISE_INVALID)
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse advertise mode, ignoring: %s", w);
+
+                config->advertise |= mode;
```

Comment:
Hmm, here `config->advertise` is updated while this function is running, but the message above says "ignoring" on error. It seems better to follow the usual style:
- if the rvalue is empty, reset `config->advertise`
- otherwise, first collect all the bits in a variable
   (if `advertise_from_string` fails, ignore that word, the same as now)
- if there's no general syntax error, update `config->advertise` by ORing the existing value with the new mask

Reply:
should do a 'continue'